tools/cli/commands/rsync.js

```diff
@@ -0,0 +1,344 @@
+import fs from 'fs/promises';
+import path from 'path';
+import process from 'process';
+import chalk from 'chalk';
+import Configstore from 'configstore';
+import execa from 'execa';
+import inquirer from 'inquirer';
+import tmp from 'tmp';
+import { projectDir } from '../helpers/install.js';
+import { listProjectFiles } from '../helpers/list-project-files.js';
+import { allProjectsByType, dirs } from '../helpers/projectHelpers.js';
+import { runCommand } from '../helpers/runCommand.js';
+
+const rsyncConfigStore = new Configstore( 'automattic/jetpack-cli/rsync' );
+
+/**
+ * Escapes dots in a string. Useful when saving destination as key in Configstore. Otherwise it tries strings with dots
+ * as nested objects.
+ *
+ * @param { string } key - String of the destination key.
+ * @returns { string } - Returns the key with escaped periods.
+ */
+function escapeKey( key ) {
+	return key.replace( /\./g, '\\.' );
+}
+
+/**
+ * Stores the destination in the configstore.
+ * Takes an optional alias arg.
+ *
+ * @param { string } pluginDestPath - Destination path to plugins.
+ * @param { string|false } alias - Alias key, if set.
+ */
+function setRsyncDest( pluginDestPath, alias = false ) {
+	const key = alias || pluginDestPath;
+	rsyncConfigStore.set( escapeKey( key ), pluginDestPath );
+}
+
+/**
+ * Entry point for the CLI.
+ * Prompts for the plugin.
+ * Prompts for the destination.
+ * Finds and uses the real wp plugin slug for dest.
+ *
+ * @param {object} argv - The argv for the command line.
+ */
+export async function rsyncInit( argv ) {
+	if ( argv.config ) {
+		await promptToManageConfig();
+		return;
+	}
+	argv = await maybePromptForPlugin( argv );
+	argv = await maybePromptForDest( argv );
+	const sourcePluginPath = projectDir( `plugins/${ argv.plugin }` ) + '/';
+	// Pull the actual plugin slug from composer.json.
+	const pluginComposerJson = await fs.readFile(
+		projectDir( `plugins/${ argv.plugin }` + '/composer.json' )
+	);
+	const wpPluginSlug = JSON.parse( pluginComposerJson ).extra[ 'wp-plugin-slug' ];
+	const finalDest = path.join( argv.dest, wpPluginSlug + '/' );
```

Comment:
So 3709c9ad improved the situation, but it'll still try to upload to `wp-content/mu-plugins/jetpack-plugin/production/jetpack/` instead of `wp-content/mu-plugins/jetpack-plugin/production/`. What do you think about doing like this?

```suggestion
	// Append plugin slug if we got a "plugins" directory, otherwise just make sure it ends in a slash.
	let finalDest;
	if ( argv.dest.match( /\/(?:mu-)?plugins\/?$/ ) ) {
		// Pull the actual plugin slug from composer.json.
		const pluginComposerJson = await fs.readFile(
			projectDir( `plugins/${ argv.plugin }/composer.json` )
		);
		const wpPluginSlug = JSON.parse( pluginComposerJson ).extra[ 'wp-plugin-slug' ];
		finalDest = path.join( argv.dest, wpPluginSlug + '/' );
	} else {
		finalDest = path.join( argv.dest, '/' );
	}
```

Reply:
Sure!